presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/delegates/EcoDisposableProperty.kt

```diff
@@ -0,0 +1,15 @@
+package com.exallium.mvvmcleanexample.presentation.utils.delegates
+
+import io.reactivex.disposables.Disposable
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class EcoDisposableProperty(private var disposable: Disposable?) : ReadWriteProperty<Any?, Disposable?> {
```

Comment:
can you make this be composite instead of a one-off? I think disposable is a general enough concept that I'd like to make ViewModel disposable.